components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/logging/APILogHandler.java

```diff
@@ -52,10 +52,15 @@ public static synchronized void logAPI(String flow, MessageContext messageContex
             return;
         }
 
-        // Get the API name
-        String apiName = ((String) messageContext.getProperty("API_TO")).split("/")[0];
+        // Get API related details
+        String[] apiTo = ((String) messageContext.getProperty("API_TO")).split("/");
```

Comment:
Shall we add a validation here so that we will not get null pointer exceptions and any index out of bound exceptions?

Reply:
added the validation.